### PR TITLE
Bug 1707928: hack/build-go.sh: use just git hashes for building

### PR DESCRIPTION
**Pull request by runcom**
**- What I did**

Do not rely on tags (as they may contain timestamps which can cause us to drift)

**- How to verify it**

**- Description for the changelog**

### CODE REVIEW

File: hack/build-go.sh [hunk 1]
```diff
@@ -17,7 +17,7 @@ cdup="$(git rev-parse --show-cdup)" && test -n "$cdup" && cd "$cdup"
 
 if [ -z ${VERSION_OVERRIDE+a} ]; then
 	echo "Using version from git..."
-	VERSION_OVERRIDE=$(git describe --abbrev=8 --dirty --always)
+	VERSION_OVERRIDE=$(git rev-parse --verify 'HEAD^{commit}')
```
Comment (cgwalters) [inline hack/build-go.sh]: Wouldn't `--verify HEAD` work too?  I had to go look up that syntax.

Are there no other repos which do something similar?  We should try to keep this in common across repos.

Comment (cgwalters) [reply hack/build-go.sh]: It's fine by me as is too.

**Comment by runcom** [comment]
damn this is failing cause we want a valid semver version

**Comment by runcom** [comment]
alright, changed to just have a new field in the version pkg that points to the git hash and use that whenever we need to compare MCC versions. @cgwalters ptal

**Comment by runcom** [comment]
gosh, this is going through another round of testing cause #701 has been merged meanwhile

**Comment by runcom** [comment]
```
annotations: {
machineconfiguration.openshift.io/generated-by-controller-version: "087bb5c36cdba30edeb2f52c69f9b2a953b800bf"
},
```
That's what we get now, and we easily point to a given MCO git commit when asked.

**Pull request merged**